app/jobs/reports/irs_weekly_summary_report.rb

```diff
@@ -0,0 +1,55 @@
+require 'csv'
+
+module Reports
+  class IrsWeeklySummaryReport < BaseReport
+    attr_reader :report_date
+    REPORT_NAME = 'irs-weekly-summary-report'
+
+    include GoodJob::ActiveJobExtensions::Concurrency
+
+    good_job_control_concurrency_with(
+      total_limit: 1,
+      key: -> { "#{REPORT_NAME}-#{arguments.first}" },
+    )
+
+    def perform(report_date)
+      @name = REPORT_NAME
+      @report_date = report_date
+
+      email = IdentityConfig.store.system_demand_report_email
+      ReportMailer.system_demand_report(
+        email: email,
+        data: generate_csv,
+        name: REPORT_NAME,
+      ).deliver_now
+
+      # save report has a predefined bucket where things get saved
+      # upload_file_to_s3_bucket can be used to define specific buckets (ie. public/private)
+      save_report(
+        REPORT_NAME,
+        generate_csv,
+        extension: 'csv',
+      )
+    end
+
+    private
+
+    # The total number of users registered with Login.gov 
+    def query_system_demand
+      User.where('created_at <= ?', report_date).count
```

Comment:
so the job scheduler calls this with `Time.zone.now` which could be at a random time, do we want this to round down to the beginning of the day (`report_date.beginning_of_day`) ? or have the job scheduler do that?

Reply:
Do we want to limit the report to users that have confirmed their email, or users that have set up at 2FA?

Reply:
Sure I can round it down in the report, no problem

Reply:
@mitchellhenke we have very little information from IRS about what is expected for these numbers. Those are great questions hopefully their response once they get back to us will add clarity but I'll follow up with them if not.